### ISSUE TITLE
Timeline Semaphore value not properly tracked when using vkSignalSemaphore

### ISSUE THREAD

**Issue by Martionex** [KhronosGroup/Vulkan-ValidationLayers#2340]
**Describe the Issue**
I'm using timeline semaphores to track the completion of compute shaders. The initial timeline semaphore value is set to 0. The signal value, indicating its completion, is 1. After completion I use `vkSignalSemaphore` to reset the timeline semaphore value back to 0, in order to reuse the same timeline semaphore for another compute shader invocation submit. However, the validation layer is incorrectly stating that the timeline semaphore value is still 1 at the time of the submit, not allowing the submit due to the requirement of the signal value to be greater than the current timeline value.

I was able to validate that the timeline value got properly reset to 0 by calling `vkGetSemaphoreCounterValue` just before calling `vkQueueSubmit`, at which point the validation error gets thrown. Note I am submitting multiple command buffers all using different timeline semaphores with a single `vkQueueSubmit`.

**Valid Usage ID**
```
Validation Error: [ VUID-VkSubmitInfo-pSignalSemaphores-03242 ] Object 0: handle = 0x9f3aab000000006e, type = VK_OBJECT_TYPE_SEMAPHORE; | MessageID = 0xdb30ee87 | VkQueueSubmit: signal value (0x1) in VkQueue 0x1b7c71c7f78[] must be greater than current timeline semaphore VkSemaphore 0x9f3aab000000006e[] value (0x1) in pSubmits[31].pSignalSemaphores[0] The Vulkan spec states: For each element of pSignalSemaphores created with a VkSemaphoreType of VK_SEMAPHORE_TYPE_TIMELINE the corresponding element of VkTimelineSemaphoreSubmitInfo::pSignalSemaphoreValues must have a value greater than the current value of the semaphore when the semaphore signal operation is executed (https://vulkan.lunarg.com/doc/view/1.2.154.1/windows/1.2-extensions/vkspec.html#VUID-VkSubmitInfo-pSignalSemaphores-03242)
```

**Environment:**
 - OS: Windows 10
 - GPU: GeForce GTX 1650
 - SDK or header version if building from repo: 1.2.154.1
 - Options enabled (synchronization, best practices, etc.): Vulkan 1.2 with VkPhysicalDeviceTimelineSemaphoreFeatures enabled

**Additional context**
Here I provide some key code fragments that bring this issue about.

Creating timeline semaphores
```
VkSemaphoreTypeCreateInfo semaphoreTypeInfo = {};
semaphoreTypeInfo.sType = VK_STRUCTURE_TYPE_SEMAPHORE_TYPE_CREATE_INFO;
semaphoreTypeInfo.semaphoreType = VK_SEMAPHORE_TYPE_TIMELINE;
semaphoreTypeInfo.initialValue = 0;

VkSemaphoreCreateInfo semaphoreInfo = {};
semaphoreInfo.sType = VK_STRUCTURE_TYPE_SEMAPHORE_CREATE_INFO;
semaphoreInfo.pNext = &semaphoreTypeInfo;
for (VkSemaphore& timelineSemaphore : m_Semaphores) {
    VK_CHECK_RESULT(vkCreateSemaphore(g_Gpu.getDevice(), &semaphoreInfo, nullptr, &timelineSemaphore));
}
```

Submitting compute tasks. Validation error occurs here.
```
VkTimelineSemaphoreSubmitInfo timelineInfo = {};
timelineInfo.sType = VK_STRUCTURE_TYPE_TIMELINE_SEMAPHORE_SUBMIT_INFO;
timelineInfo.signalSemaphoreValueCount = 1;
timelineInfo.pSignalSemaphoreValues = &TIMELINE_FINISHED_VALUE; // = 1
...
for (uint32_t i = 0; i < tasksToSubmit.size(); ++i) {
	...
	VkSubmitInfo& submitInfo = submitInfos[i];
	submitInfo.sType = VK_STRUCTURE_TYPE_SUBMIT_INFO;
	submitInfo.commandBufferCount = 1;
	submitInfo.pCommandBuffers = &m_CommandBuffers[taskIndex];
	submitInfo.signalSemaphoreCount = 1;
	submitInfo.pSignalSemaphores = &m_Semaphores[taskIndex];
	submitInfo.pNext = &timelineInfo;

	// vkGetSemaphoreCounterValue is always 0 here
}
...
// Validation error here
VK_CHECK_RESULT(vkQueueSubmit(g_Gpu.getComputeQueue(), static_cast<uint32_t>(tasksToSubmit.size()), submitInfos.data(), nullptr));
```

Resetting timeline semaphore back to 0. This is done every time a submit finishes by reading the timeline value, which will be 1 at the time of completion. It seems as if the validation layers do not realise this change took place, as it still believes the timeline semaphore value to be 1 even though `vkGetSemaphoreCounterValue` says otherwise.
```
VkSemaphoreSignalInfo signalInfo = {};
signalInfo.sType = VK_STRUCTURE_TYPE_SEMAPHORE_SIGNAL_INFO;
signalInfo.semaphore = semaphore;
signalInfo.value = 0;
vkSignalSemaphore(g_Gpu.getDevice(), &signalInfo);
```

**Comment by sfricke-samsung** [comment]
I think I fixed this in
https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/2272

**Comment by mark-lunarg** [comment]
This PR (#2275) was merged in the 1.2.158 timeframe, a few weeks after the 1.2.154 SDK.  @martionex, if you could build the repo to verify, that'd be great.  Alternatively, you might try the [Windows build artifacts here](https://ci.appveyor.com/project/Khronoswebmaster/vulkan-validationlayers/builds/36473781/job/wmp7qi91vk8ksbir/artifacts) and let us know if this issue has been sufficiently addressed.  Thanks!

**Comment by jeremyg-lunarg** [comment]
Closing because AFAICT the problem has been fixed for quite some time.  Note that you cannot ever reset a timeline semaphores value, as VUID-VkSubmitInfo-pSignalSemaphores-03242 states. The only fix was to correct the values printed in the error.
